### PR TITLE
[Enhancement] make map keys unique and implement distinct_map_keys() for external tables

### PR DESCRIPTION
**Pull request by fzhedu**
## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

## Problem Summary(Required):

remove duplicated keys for maps in internal tables, and achieve that for external tables by `distinct_map_keys()`

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [x] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/exprs/map_functions.cpp [hunk 1]
```diff
@@ -285,4 +285,74 @@ void MapFunctions::_filter_map_items(const MapColumn* src_column, const ColumnPt
     dest_column->values_column()->append_selective(src_column->values(), indexes);
 }
 
+StatusOr<ColumnPtr> MapFunctions::distinct_map_keys(FunctionContext* context, const Columns& columns) {
```
Comment (Seaven) [inline be/src/exprs/map_functions.cpp:288]: `distinct_map` is more exact?

Comment (fzhedu) [reply be/src/exprs/map_functions.cpp:288]: not really, distinct_map should remove identical map with the same keys as well as values.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PolymorphicFunctionAnalyzer.java|1|2|50.00%|[177]|

**Review by Seaven — APPROVED** [review]

File: be/src/exprs/map_functions.cpp [hunk 2]
```diff
@@ -285,4 +285,74 @@ void MapFunctions::_filter_map_items(const MapColumn* src_column, const ColumnPt
...
+                if (hash[j] == hash[k] && keys->equals(j, *keys, k)) {
+                    filter[j] = 0;
+                    has_duplicated_keys = true;
+                    break;
+                }
+            }
+            new_offset += filter[j];
+        }
+        offsets_vec.push_back(new_offset);
+    }
```
Comment (silverbullet233) [inline be/src/exprs/map_functions.cpp:330]: seems a log of codes are duplicated in `MapColumn:: remove_duplicated_keys`, do we have a way to unify them?

Comment (fzhedu) [reply be/src/exprs/map_functions.cpp:330]: not really, `remove_duplicated_keys ` change original map, but this function return a new map, used in different cases.

**Review by silverbullet233 — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**